plugins/modules/aci_netflow_exporter_policy.py

```diff
@@ -0,0 +1,446 @@
+        description=dict(type="str", aliases=["descr"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "netflow_exporter_policy"]],
+            ["state", "present", ["tenant", "netflow_exporter_policy", "destination_address", "destination_port"]],
+        ],
+        mutually_exclusive=[["associated_epg", "associated_extepg"]],
+    )
+
+    tenant = module.params.get("tenant")
+    description = module.params.get("description")
+    netflow_exporter_policy = module.params.get("netflow_exporter_policy")
+    dscp = module.params.get("dscp")
+    destination_address = module.params.get("destination_address")
+    destination_port = module.params.get("destination_port")
+    source_ip_type = MATCH_SOURCE_IP_TYPE_NETFLOW_EXPORTER_MAPPING.get(module.params.get("source_ip_type"))
+    custom_source_address = module.params.get("custom_source_address")
+    associated_epg = module.params.get("associated_epg")
+    associated_extepg = module.params.get("associated_extepg")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    child_classes = ["netflowRsExporterToCtx", "netflowRsExporterToEPg"]
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="netflowExporterPol",
+            aci_rn="exporterpol-{0}".format(netflow_exporter_policy),
+            module_object=netflow_exporter_policy,
+            target_filter={"name": netflow_exporter_policy},
+        ),
+        child_classes=child_classes,
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        if associated_epg is not None:
+            if all(value is None for value in associated_epg.values()) and isinstance(aci.existing, list) and len(aci.existing) > 0:
+                for child in aci.existing[0].get("netflowExporterPol", {}).get("children", {}):
+                    if child.get("netflowRsExporterToCtx"):
+                        child_configs.extend([dict(netflowRsExporterToCtx=dict(attributes=dict(status="deleted")))])
+                    elif child.get("netflowRsExporterToEPg"):
+                        child_configs.extend([dict(netflowRsExporterToEPg=dict(attributes=dict(status="deleted")))])
+            elif all(value is not None for value in associated_epg.values()):
+                associated_tenant = associated_epg.get("tenant")
+                child_configs = [
+                    dict(netflowRsExporterToCtx=dict(attributes=dict(tDn="uni/tn-{0}/ctx-{1}".format(associated_tenant, associated_epg.get("vrf"))))),
```

Comment:
I think you need to specify VRF and EPG separately. You are unable to select the VRF in common tenant now for. Which is a possibility I think.

Reply:
Probably best to check options on external EPG also